fourm/data/modality_transforms.py

```diff
@@ -867,8 +947,129 @@ def postprocess(self, bboxes):
+            "num_instances": 5,
+            "image_height": 512,
+            "image_width": 906,
+            "instances": [
+                {
+                    "boxes": [
+                        0.4229210317134857,
+                        0.00020096010121051222,
+                        0.5715101361274719,
+                        0.13699540495872498
+                    ],
+                    "score": 0.9029952883720398,
+                    "class_id": 74,
+                    "class_name": "clock",
+                    "segmentation": [
+                        [
+                            0.5055187637969095,
+                            0.1337890625,
+                            ...
+                        ]
+                    ]
+                },
+                {
+                    "boxes": [
+                        ...
+                    ],
+                    ...
+                },
+                    ...
+            ]
+        },
+        # FRAME 1 Bounding boxes
+        {
+            "num_instances": 5,
+            "image_height": 512,
+            "image_width": 906,
+            "instances": [
+                ...,
+            ],
+            ...
+        }
+    ]
+
+    Input: path to a List[dicts] representing bounding boxes representations for each frame.
+    Output: a str representing the bounding boxes for each frame in the video, separated by special frame tokens to indicate which frame each bounding box representation corresponds to.
+    """
+
+    def load(self, path):
+        """
+        Load jsonl file containing bounding box representations per frame.
+        """
+        with jsonlines.open(path, "r") as jsonl_f:
+            bbs_per_frame = [obj for obj in jsonl_f]
+
+        return bbs_per_frame
+
+    def preprocess(self, sample: List[dict]) -> List[List[tuple]]:
+        """
+        Given a list of dicts (each element in the list is a frame, each dict is a representation of bounding boxes for that frame),
+        convert the instances within each frame's bounding boxes representation.
+        # TODO: any issue with returning a list of lists here? We sorta might break the type consistency of the class? I'm not sure how robust it is.
+            # ^ Actually I think this is fine because it just gets passed into the image_augment function, so as long as that's implemented correspondingly we're good.
+
+        Returns:
+            List of lists of tuples.
+            The outermost list represents each frame.
+            The middle list represents each bounding box instances in that frame.
+            The innermost tuple is the actual representation of a bounding box instance (of the form [xmin, ymin, xmax, ymax, class_name, score]).
+        """
+        instances = [frame["instances"] for frame in sample]
+        return self.convert_detection_instance(instances)
+
+    def image_augment(
+        self,
+        bboxes_per_frame: List[List[Tuple]],
+        crop_coords: Tuple,
```

Comment:
No rand_aug_idx for this because the other parameters actually specify which aug to apply

Reply:
so we have the same crop for each frame already!